app/static/src/components/code/CodeEditor.vue

```diff
@@ -1,7 +1,7 @@
 <template>
     <div>
         <button
-            v-if="defaultCodeExists"
+            v-if="defaultCodeExists && !STATIC_BUILD"
```

Comment:
I feel like it would maybe be a bit nicer if the components themselves didn't know directly about STATIC_BUILD and read a getter from the store to say if code should be resettable etc.

Reply:
i can see that in this case perhaps, but then places like `WodinSession.vue` we have code in a watcher that relies on the static build variable, in `BasicApp.vue` and other app types we have a switch on whether "Options" is the tab shown initially or not, and in some places we just have a `v-if` on only the static build variable

so perhaps in the app types we can have shared code, either a getter or a function that tells us what the initial tab is, but for a `v-if` on just the static build variable im not sure a getter or a function does anything for us, itll just wrap static build in another layer

yh im just not sure how we have getters that cover all these slightly different ways of using static build without basically creating a different getter for each of these use cases, at which point we may as well use it in the component

Reply:
yeah, fair point. But as you say for the lower level components like this one, maybe it should just be a prop.

Reply:
if i were to pull this condition out then i would need to compute `defaultCodeExists` in the parent component which is not a problem, but the question is, should this component still be responsible for resetting the code (it currently resets the code itself)?
* on one hand it feels weird to toggle visibility of the reset button in the parents and then the child control the actual action of resetting
* on the other hand i think it would be messier to pull out the functionality of resetting the code from this component because it needs the editor instance

i cant quite think of a better idea yet

i have however taken out the `STATIC_BUILD` from `BasicApp.vue`, `FitApp.vue` and `StochasticApp.vue` and used the prop idea there! because i could concentrate that logic in the `router.ts` file

Reply:
I suppose one thing you could do would be to have something at the top level of the store or app be opinionated about how `STATIC_BUILD` being true should impact how the app should interpret AppConfig e.g. it could set `readOnlyCode` to true even if wasn't defined as such in the config it loaded. So then the components don't care that it's a static build, as far as they're concerned it's just any read only app. 

That approach doesnt work for removing the Session stuff, as that is a static build special.

Reply:
hmmm thats a good idea actually! readOnlyCode quirks will be solved! yh think the apiService one, sessions page one and wodinSession ones will probably remain, but thats not too bad!